### ISSUE TITLE
Setting 'force-tenant-connection-of-models' for external library model not working

### ISSUE THREAD

**Issue by Bibendus83** [hyn/multi-tenant#913]
__Description__

I tried using the setting `force-tenant-connection-of-models` in `tenancy.php` to force the tenant connection on an external library `Model` but the setting is ignored.
```php
'force-tenant-connection-of-models' => [
    \QCod\Settings\Setting\Setting::class,
],
```
The library I'm using using is `qcod/laravel-app-settings` and I'm using it to easily implement custom settings for each tenant.
When I try to use it I get this error:
```
Illuminate\Database\QueryException thrown with message "SQLSTATE[42S02]: Base table or view not found: 1146 Table 'tenancy.settings' doesn't exist (SQL: select `val`, `name` from `settings` where `group` = default)"

Stacktrace:
#79 Illuminate\Database\QueryException in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:664
#78 Doctrine\DBAL\Driver\PDOException in /shared/httpd/my-app/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:63
#77 PDOException in /shared/httpd/my-app/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:61
#76 PDO:prepare in /shared/httpd/my-app/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:61
#75 Doctrine\DBAL\Driver\PDOConnection:prepare in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:326
#74 Illuminate\Database\Connection:Illuminate\Database\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:657
#73 Illuminate\Database\Connection:runQueryCallback in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:624
#72 Illuminate\Database\Connection:run in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Connection.php:333
#71 Illuminate\Database\Connection:select in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2130
#70 Illuminate\Database\Query\Builder:runSelect in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2324
#69 Illuminate\Database\Query\Builder:Illuminate\Database\Query\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2604
#68 Illuminate\Database\Query\Builder:onceWithColumns in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2326
#67 Illuminate\Database\Query\Builder:pluck in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:717
#66 Illuminate\Database\Eloquent\Builder:pluck in /shared/httpd/my-app/vendor/qcod/laravel-settings/src/Setting/SettingEloquentStorage.php:34
#65 QCod\Settings\Setting\SettingEloquentStorage:QCod\Settings\Setting\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Cache/Repository.php:418
#64 Illuminate\Cache\Repository:rememberForever in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php:357
#63 Illuminate\Cache\CacheManager:__call in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php:239
#62 Illuminate\Support\Facades\Facade:__callStatic in /shared/httpd/my-app/vendor/qcod/laravel-settings/src/Setting/SettingEloquentStorage.php:35
#61 QCod\Settings\Setting\SettingEloquentStorage:all in /shared/httpd/my-app/vendor/qcod/laravel-settings/src/Setting/SettingEloquentStorage.php:43
#60 QCod\Settings\Setting\SettingEloquentStorage:get in /shared/httpd/my-app/vendor/qcod/laravel-app-settings/src/Setting/AppSettings.php:60
#59 QCod\AppSettings\Setting\AppSettings:get in /shared/httpd/my-app/vendor/qcod/laravel-app-settings/src/helpers.php:24
#58 setting in /shared/httpd/my-app/app/Http/Middleware/InitJavascriptConstants.php:27
#57 App\Http\Middleware\InitJavascriptConstants:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#56 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#55 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:41
#54 Illuminate\Routing\Middleware\SubstituteBindings:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#53 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#52 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:75
#51 Illuminate\Foundation\Http\Middleware\VerifyCsrfToken:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#50 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#49 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49
#48 Illuminate\View\Middleware\ShareErrorsFromSession:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#47 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#46 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:56
#45 Illuminate\Session\Middleware\StartSession:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#44 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#43 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37
#42 Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#41 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#40 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:66
#39 Illuminate\Cookie\Middleware\EncryptCookies:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#38 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#37 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:104
#36 Illuminate\Pipeline\Pipeline:then in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Router.php:682
#35 Illuminate\Routing\Router:runRouteWithinStack in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Router.php:657
#34 Illuminate\Routing\Router:runRoute in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Router.php:623
#33 Illuminate\Routing\Router:dispatchToRoute in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Router.php:612
#32 Illuminate\Routing\Router:dispatch in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:176
#31 Illuminate\Foundation\Http\Kernel:Illuminate\Foundation\Http\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:30
#30 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php:65
#29 Barryvdh\Debugbar\Middleware\InjectDebugbar:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#28 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#27 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/app/Http/Middleware/ForceHttpsProtocol.php:17
#26 App\Http\Middleware\ForceHttpsProtocol:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#25 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#24 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/fideloper/proxy/src/TrustProxies.php:57
#23 Fideloper\Proxy\TrustProxies:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#22 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#21 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21
#20 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#19 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#18 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:21
#17 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#16 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#15 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php:27
#14 Illuminate\Foundation\Http\Middleware\ValidatePostSize:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#13 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#12 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php:62
#11 Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#10 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#9 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/hyn/multi-tenant/src/Middleware/EagerIdentification.php:29
#8 Hyn\Tenancy\Middleware\EagerIdentification:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#7 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#6 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/hyn/multi-tenant/src/Middleware/HostnameActions.php:75
#5 Hyn\Tenancy\Middleware\HostnameActions:handle in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:163
#4 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#3 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:104
#2 Illuminate\Pipeline\Pipeline:then in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:151
#1 Illuminate\Foundation\Http\Kernel:sendRequestThroughRouter in /shared/httpd/my-app/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:116
#0 Illuminate\Foundation\Http\Kernel:handle in /shared/httpd/my-app/public/index.php:55
```

---
__Information__

- hyn/multi-tenant version: 5.4.5
- laravel version: 5.8
- database driver and version: MariaDB 10.3
- webserver software and version: Apache 2.4
- php version: 7.3

__tenancy.php config__

```php
<?php

/*
 * This file is part of the hyn/multi-tenant package.
 *
 * (c) Daniël Klabbers <daniel@klabbers.email>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @see https://tenancy.dev
 * @see https://github.com/hyn/multi-tenant
 */

use Hyn\Tenancy\Database\Connection;

return [
    /**
     * Random key used for tenant database user password
     */
    'key' => env('TENANCY_KEY', env('APP_KEY')),

    'models' => [
        /**
         * Specify different models to be used for the global, system database
         * connection. These are also used in their relationships. Models
         * used have to implement their respective contracts and
         * either extend the SystemModel or use the trait
         * UsesSystemConnection.
         */

        // Must implement \Hyn\Tenancy\Contracts\Hostname
        'hostname' => \Hyn\Tenancy\Models\Hostname::class,

        // Must implement \Hyn\Tenancy\Contracts\Website
        'website' => \Hyn\Tenancy\Models\Website::class
    ],
    /**
     * The package middleware. Removing a middleware here will disable it.
     * You can of course extend/replace them or add your own.
     */
    'middleware' => [
        // The eager identification middleware.
        \Hyn\Tenancy\Middleware\EagerIdentification::class,

        // The hostname actions middleware (redirects, https, maintenance).
        \Hyn\Tenancy\Middleware\HostnameActions::class,
    ],
    'website' => [
        /**
         * Each website has a short random hash that identifies this entity
         * to the application. By default this id is randomized and fully
         * auto-generated. In case you want to force your own logic for
         * when you need to have a better overview of the complete
         * tenant folder structure, disable this and implement
         * your own id generation logic.
         */
        'disable-random-id' => false,

        /**
         * The random Id generator is responsible for creating the hash as mentioned
         * above. You can override what generator to use by modifying this value
         * in the configuration.
         *
         * @warn This won't work if disable-random-id is true.
         */
        'random-id-generator' => Hyn\Tenancy\Generators\Uuid\ShaGenerator::class,

        /**
         * Enable this flag in case you're using a driver that does not support
         * database username or database name with a length of more than 32 characters.
         *
         * This should be enabled for MySQL, but not for MariaDB and PostgreSQL.
         */
        'uuid-limit-length-to-32' => env('LIMIT_UUID_LENGTH_32', false),

        /**
         * Specify the disk you configured in the filesystems.php file where to store
         * the tenant specific files, including media, packages, routes and other
         * files for this particular website.
         *
         * @info If not set, will revert to the default filesystem.
         * @info If set to false will disable all tenant specific filesystem auto magic
         *       like the config, vendor overrides.
         */
        'disk' => null,

        /**
         * Automatically generate a tenant directory based on the random id of the
         * website. Uses the above disk to store files to override system-wide
         * files.
         *
         * @info set to false to disable.
         */
        'auto-create-tenant-directory' => true,

        /**
         * Automatically rename the tenant directory when the random id of the
         * website changes. This should not be too common, but in case it happens
         * we automatically want to move files accordingly.
         *
         * @info set to false to disable.
         */
        'auto-rename-tenant-directory' => true,

        /**
         * Automatically deletes the tenant specific directory and all files
         * contained within.
         *
         * @see
         * @info set to true to enable.
         */
        'auto-delete-tenant-directory' => false,

        /**
         * Time to cache websites in minutes. Set to false to disable.
         */
        'cache' => 10,
    ],
    'hostname' => [
        /**
         * If you want the multi tenant application to fall back to a default
         * hostname/website in case the requested hostname was not found
         * in the database, complete in detail the default hostname.
         *
         * @warn this must be a FQDN, these have no protocol or path!
         */
        'default' => env('TENANCY_DEFAULT_HOSTNAME'),
        /**
         * The package is able to identify the requested hostname by itself,
         * disable to get full control (and responsibility) over hostname
         * identification. The hostname identification is needed to
         * set a specific website as currently active.
         *
         * @see src/Jobs/HostnameIdentification.php
         */
        'auto-identification' => env('TENANCY_AUTO_HOSTNAME_IDENTIFICATION', true),

        /**
         * In case you want to have the tenancy environment set up early,
         * enable this flag. This will run the tenant identification
         * inside a middleware. This will eager load tenancy.
         *
         * A good use case is when you have set "tenant" as the default
         * database connection.
         */
        'early-identification' => env('TENANCY_EARLY_IDENTIFICATION', true),

        /**
         * Abort application execution in case no hostname was identified. This will throw a
         * 404 not found in case the tenant hostname was not resolved.
         */
        'abort-without-identified-hostname' => env('TENANCY_ABORT_WITHOUT_HOSTNAME', false),

        /**
         * Time to cache hostnames in minutes. Set to false to disable.
         */
        'cache' => env('TENANCY_CACHE_TIME', 10),

        /**
         * Automatically update the app.url configured inside Laravel to match
         * the tenant FQDN whenever a hostname/tenant was identified.
         *
         * This will resolve issues with password reset mails etc using the
         * correct domain.
         */
        'update-app-url' => env('TENANCY_UPDATE_APP_URL', false),

        /**
         * Root default FQDN where all subdomains are created by default
         */
        'root-fqdn' => env('TENANCY_ROOT_FQDN', false),
    ],
    'db' => [
        /**
         * The default connection to use; this overrules the Laravel database.default
         * configuration setting. In Laravel this is normally configured to 'mysql'.
         * You can set a environment variable to override the default database
         * connection to - for instance - the tenant connection 'tenant'.
         */
        'default' => env('TENANCY_DEFAULT_CONNECTION'),
        /**
         * Used to give names to the system and tenant database connections. By
         * default we configure 'system' and 'tenant'. The tenant connection
         * is set up automatically by this package.
         *
         * @see src/Database/Connection.php
         * @var system-connection-name The database connection name to use for the global/system database.
         * @var tenant-connection-name The database connection name to use for the tenant database.
         */
        'system-connection-name' => env('TENANCY_SYSTEM_CONNECTION_NAME', Connection::DEFAULT_SYSTEM_NAME),
        'tenant-connection-name' => env('TENANCY_TENANT_CONNECTION_NAME', Connection::DEFAULT_TENANT_NAME),

        /**
         * The tenant division mode specifies to what database websites will be
         * connecting. The default setup is to use a new database per tenant.
         * If using PostgreSQL, a new schema per tenant in the same database can
         * be setup, by optionally setting division mode to 'schema'.
         * In case you prefer to use the same database with a table prefix,
         * set the mode to 'prefix'.
         * To implement a custom division mode, set this to 'bypass'.
         *
         * @see src/Database/Connection.php
         */
        'tenant-division-mode' => env('TENANCY_DATABASE_DIVISION_MODE', 'database'),

        /**
         * The database password generator takes care of creating a valid hashed
         * string used for tenants to connect to the specific database. Do
         * note that this will only work in 'division modes' that set up
         * a connection to a separate database.
         */
        'password-generator' => Hyn\Tenancy\Generators\Database\DefaultPasswordGenerator::class,

        /**
         * The tenant migrations to be run during creation of a tenant. Specify a directory
         * to run the migrations from. If specified these migrations will be executed
         * whenever a new tenant is created.
         *
         * @info set to false to disable auto migrating.
         *
         * @warn this has to be an absolute path, feel free to use helper methods like
         * base_path() or database_path() to set this up.
         */
        'tenant-migrations-path' => database_path('migrations/tenant'),

        /**
         * The default Seeder class used on newly created databases and while
         * running artisan commands that fire seeding.
         *
         * @info requires tenant-migrations-path in order to seed newly created websites.
         * @info seeds stored in `database/seeds/tenants` need to be configured in your composer.json classmap.
         *
         * @warn specify a valid fully qualified class name.
         */
        // 'tenant-seed-class' => false,
//      eg an admin seeder under `app/Seeders/AdminSeeder.php`:
       'tenant-seed-class' => AdminTableSeeder::class,

        /**
         * Automatically generate a tenant database based on the random id of the
         * website.
         *
         * @info set to false to disable.
         */
        'auto-create-tenant-database' => true,

        /**
         * Automatically generate the user needed to access the database.
         *
         * @info Useful in case you use root or another predefined user to access the
         *       tenant database.
         * @info Only creates in case tenant databases are set to be created.
         *
         * @info set to false to disable.
         */
        'auto-create-tenant-database-user' => true,

        /**
         * Set of database privileges to give to the tenant database user.
         *
         * @info Useful in case your database restricts the privileges you
         *       can set (for example AWS RDS).
         * @info These privileges are only used in case tenant database users
         *       are set to be created.
         *
         * @info null by default means "ALL PRIVILEGES". Override with a list
         *       of privileges as a string, e.g. 'SELECT, UPDATE'.
         */
        'tenant-database-user-privileges' => null,

        /**
         * Automatically rename the tenant database when the random id of the
         * website changes. This should not be too common, but in case it happens
         * we automatically want to move databases accordingly.
         *
         * @info set to false to disable.
         */
        'auto-rename-tenant-database' => true,

        /**
         * Automatically deletes the tenant specific database and all data
         * contained within.
         *
         * @info set to true to enable.
         */
        'auto-delete-tenant-database' => env('TENANCY_DATABASE_AUTO_DELETE', false),

        /**
         * Automatically delete the user needed to access the tenant database.
         *
         * @info Set to false to disable.
         * @info Only deletes in case tenant database is set to be deleted.
         */
        'auto-delete-tenant-database-user' => env('TENANCY_DATABASE_AUTO_DELETE_USER', false),

        /**
         * Define a list of classes that you wish to force onto the tenant or system connection.
         * The connection will be forced when the Model has booted.
         *
         * @info Useful for overriding the connection of third party packages.
         */
        'force-tenant-connection-of-models' => [
            \QCod\Settings\Setting\Setting::class,
        ],
        'force-system-connection-of-models' => [
//            App\User::class
        ],
    ],

    /**
     * Global tenant specific routes.
     * Making it easier to distinguish between landing and tenant routing.
     *
     * @info only works with `tenancy.hostname.auto-identification` or identification happening
     *       before the application is booted (eg inside middleware or the register method of
     *       service providers).
     */
    'routes' => [
        /**
         * Routes file to load whenever a tenant was identified.
         *
         * @info Set to false or null to disable.
         */
        'path' => base_path('routes/tenants.php'),

        /**
         * Set to true to flush all global routes before setting the routes from the
         * tenants.php routes file.
         */
        'replace-global' => false,
    ],

    /**
     * Folders configuration specific per tenant.
     * The following section relates to configuration to files inside the tenancy/<uuid>
     * tenant directory.
     */
    'folders' => [
        'config' => [
            /**
             * Merge configuration files from the config directory
             * inside the tenant directory with the global configuration files.
             */
            'enabled' => true,

            /**
             * List of configuration files to ignore, preventing override of crucial
             * application configurations.
             */
            'blacklist' => ['database', 'tenancy', 'webserver'],
        ],
        'routes' => [
            /**
             * Allows adding and overriding URL routes inside the tenant directory.
             */
            'enabled' => true,

            /**
             * Prefix all tenant routes.
             */
            'prefix' => null,
        ],
        'trans' => [
            /**
             * Allows reading translation files from a trans directory inside
             * the tenant directory.
             */
            'enabled' => true,

            /**
             * Will override the global translations with the tenant translations.
             * This is done by overriding the laravel default translator with the new path.
             */
            'override-global' => true,

            /**
             * In case you disabled global override, specify a namespace here to load the
             * tenant translation files with.
             */
            'namespace' => 'tenant',
        ],
        'vendor' => [
            /**
             * Allows using a custom vendor (composer driven) folder inside
             * the tenant directory.
             */
            'enabled' => true,
        ],
        'media' => [
            /**
             * Mounts the assets directory with (static) files for public use.
             */
            'enabled' => true,
        ],
        'views' => [
            /**
             * Enables reading views from tenant directories.
             */
            'enabled' => true,

            /**
             * Specify a namespace to use with which to load the views.
             *
             * @eg setting `tenant` will allow you to use `tenant::some.blade.php`
             * @info set to null to add to the global namespace.
             */
            'namespace' => null,

            /**
             * If `namespace` is set to null (thus using the global namespace)
             * make it override the global views. Disable by setting to false.
             */
            'override-global' => true,
        ]
    ]
];
```

__webserver.php config__

```php
<?php
return [

    /**
     * Apache2 is one of the most widely adopted webserver packages available.
     *
     * @see http://httpd.apache.org/docs/
     * @see https://www.digitalocean.com/community/tutorials/how-to-install-linux-apache-mysql-php-lamp-stack-on-ubuntu
     */
    'apache2' => [
        /**
         * Whether the integration with Apache2 is currently active.
         */
        'enabled' => false,

        /**
         * Define the ports of your Apache service.
         */
        'ports' => [
            /**
             * HTTP, non-SSL port.
             *
             * @default 80
             */
            'http' => 80,
            /**
             * HTTPS, SSL port.
             *
             * @default 443
             */
            'https' => 443
        ],

        /**
         * The generator taking care of hooking into the Apache services and files.
         */
        'generator' => \Hyn\Tenancy\Generators\Webserver\Vhost\ApacheGenerator::class,

        /**
         * The view that holds the vhost configuration template.
         */
        'view' => 'tenancy.generators::webserver.apache.vhost',

        /**
         * Specify the disk you configured in the filesystems.php file where to store
         * the tenant vhost configuration files.
         *
         * @info If not set, will revert to the default filesystem.
         */
        'disk' => null,

        'paths' => [

            /**
             * Location where vhost configuration files can be found.
             */
            'vhost-files' => [
                '/etc/apache2/sites-enabled/'
            ],

            /**
             * Actions to run to work with the Apache2 service.
             */
            'actions' => [
                /**
                 * Action that asserts Apache2 is installed.
                 */
                'exists' => '/etc/init.d/apache2',
                /**
                 * Action to run to test the apache configuration.
                 *
                 * @set to a boolean to force the response of the test command.
                 * @info true succeeds, false fails
                 */
                'test-config' => 'apache2ctl -t',
                /**
                 * Action to run to reload the apache service.
                 *
                 * @info set to null to disable reloading.
                 */
                'reload' => 'apache2ctl graceful'
            ]
        ]
    ],

    /**
     * Nginx webserver support.
     *
     * @see http://nginx.org
     */
    'nginx' => [
        /**
         * Whether the integration with nginx is currently active.
         */
        'enabled' => false,

        /**
         * The php sock to be used.
         */
        'php-sock' => 'unix:/var/run/php/php7.3-fpm.sock',

        /**
         * Define the ports of your nginx service.
         */
        'ports' => [
            /**
             * HTTP, non-SSL port.
             *
             * @default 80
             */
            'http' => 80,
            /**
             * HTTPS, SSL port.
             *
             * @default 443
             */
            'https' => 443
        ],

        /**
         * The generator taking care of hooking into the nginx services and files.
         */
        'generator' => \Hyn\Tenancy\Generators\Webserver\Vhost\NginxGenerator::class,

        /**
         * The view that holds the vhost configuration template.
         */
        'view' => 'tenancy.generators::webserver.nginx.vhost',

        /**
         * Specify the disk you configured in the filesystems.php file where to store
         * the tenant vhost configuration files.
         *
         * @info If not set, will revert to the default filesystem.
         */
        'disk' => null,

        'paths' => [

            /**
             * Location where vhost configuration files can be found.
             */
            'vhost-files' => [
                '/etc/nginx/sites-enabled/'
            ],

            /**
             * Actions to run to work with the Nginx service.
             */
            'actions' => [
                /**
                 * Action that asserts nginx is installed.
                 */
                'exists' => '/etc/init.d/nginx',
                /**
                 * Action to run to test the nginx configuration.
                 *
                 * @info set to a boolean to force the response of the test command.
                 *  true succeeds, false fails
                 */
                'test-config' => '/etc/init.d/nginx configtest',
                /**
                 * Action to run to reload the nginx service.
                 *
                 * @info set to null to disable reloading.
                 */
                'reload' => '/etc/init.d/nginx reload'
            ]
        ]
    ]
];
```

---

__.env file__
```env
APP_NAME="My App"
APP_ENV=local
APP_KEY=XXXXXXXXXXXXXXX
APP_DEBUG=true
APP_URL=https://www.my-app.loc:4430/
ASSET_URL=" "

LOCALE=it

LOG_CHANNEL=stack

DB_CONNECTION=system
DB_HOST=172.16.238.10
DB_PORT=3306
DB_DATABASE=tenant
DB_USERNAME=root
DB_PASSWORD=

# TENANCY_DEFAULT_CONNECTION=tenant
TENANCY_DRIVER=mysql
TENANCY_HOST=172.16.238.10
TENANCY_PORT=3306
TENANCY_DATABASE=tenancy
TENANCY_USERNAME=root
TENANCY_PASSWORD=XXXXXXXXXX
TENANCY_CACHE_TIME=false
TENANCY_AUTO_HOSTNAME_IDENTIFICATION=true
TENANCY_ABORT_WITHOUT_HOSTNAME=true
TENANCY_UPDATE_APP_URL=false    # setting it to true causes issues with url() and ports
TENANCY_DATABASE_DIVISION_MODE=database
TENANCY_ROOT_FQDN=my-app.loc
LIMIT_UUID_LENGTH_32=false

BROADCAST_DRIVER=log
CACHE_DRIVER=array
CACHE_EXPIRE=-1
QUEUE_CONNECTION=database
QUEUE_DRIVER=sync   # DEV
SESSION_DRIVER=file
SESSION_LIFETIME=120
...
```

**Comment by Bibendus83** [comment]
I noticed there is another open ticket about this #751

**Comment by fletch3555** [comment]
Let's close this and just track it under #751

**Comment by Bibendus83** [comment]
As a side note, for future reference, I was able to make the library `qcod/laravel-app-settings` work with jobs flawlessly.
First I noticed I had a issue with caching, even if each tenant has its settings table I had to define a different cache key for each one specifying a cache tenant identifier in `app_settings.php`
```
    'setting_group' => function() {
        $website   = app(\Hyn\Tenancy\Environment::class)->tenant();
        if (!$website) {
            throw new \Exception("Tenant not set while trying to read app setting value");
        }
        return "tenant-".$website->id;
    }
```

As a second thing, considering the library model `Setting.php` can't be extended to add the  `UsesTenantConnection` trait, I was able to extend it anyway with some work.
I disabled laravel auto discovery and added my custom `SettingsServiceProvider` class in `app.php`.
Following the chain of classess I was able then to force the use of my custom `Setting.php` model class where I addedd the `UsesTenantConnection` trait.
